eth/eventservices/roundservice.go

```diff
@@ -75,18 +79,77 @@ func (s *RoundsService) tryInitializeRound() (bool, error) {
 	}
 
 	if lastInitializedRound.Cmp(currentRound) == -1 {
-		tx, err := s.client.InitializeRound()
+		shouldInitialize, err := s.shouldInitializeRound(blkNum, blkHash)
 		if err != nil {
 			return false, err
 		}
 
-		err = s.client.CheckTx(tx)
-		if err != nil {
-			return false, err
-		}
+		if shouldInitialize {
+			glog.Infof("New round - preparing to initialize round to join active set")
 
-		glog.Infof("Initialized round %v", currentRound)
+			tx, err := s.client.InitializeRound()
+			if err != nil {
+				return false, err
+			}
+
+			err = s.client.CheckTx(tx)
+			if err != nil {
+				return false, err
+			}
+
+			glog.Infof("Initialized round %v", currentRound)
+		}
 	}
 
 	return true, nil
 }
+
+func (s *RoundsService) shouldInitializeRound(blkNum *big.Int, blkHash common.Hash) (bool, error) {
+	// Check to initialize round only in multiples of BlocksToWait blocks
+	// to make sure a previous initializeRound call by someone else is processed
+	blkNumMod := new(big.Int).Mod(blkNum, BlocksToWait)
```

Comment:
Rather than doing `blkNum`, should we do `blkNum - (block # the round started)`, so this way this always returns `0` for the first block in the round. Otherwise we could have up to a 4 block delay before the round is started and all transactions would fail in the meantime.